src/mito2/src/worker.rs

```diff
@@ -638,6 +649,26 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             let _ = sender.send(SetReadonlyResponse::NotFound);
         }
     }
+
+    async fn edit_region(&self, region_id: RegionId, edit: RegionEdit) -> Result<()> {
+        let region = self
+            .regions
+            .get_region(region_id)
+            .context(RegionNotFoundSnafu { region_id })?;
+
+        for file_meta in edit.files_to_add.iter() {
+            let is_exist = region.access_layer.is_exist(file_meta).await?;
+            ensure!(
+                is_exist,
+                EditRegionSnafu {
+                    reason: format!("file '{}' not exist", file_meta.file_id)
+                }
+            );
+        }
+
+        // Applying region edit directly has nothing to do with memtables (at least for now).
+        region.apply_edit(edit, &[]).await
```

Comment:
Do we need to update the last_entry_id and committed_sequence of the region? Otherwise, they will be less than the flushed entry id and sequence.

Maybe we should add a new method to apply edits from outside.

Reply:
I don't think it's needed because the region editing is only used for adding ssts directly, now. Not related to WAL and memtable. But the RegionEdit carries other fields that are considered inside `apply_edit`. So maybe we can simply check the RegionEdit here to ensure there's only `files_to_add`?

Reply:
Let's add these checks. I think removing files is acceptable. But we should document in this method that if we use region edit to ingest data, then we MUST not write data to the region directly.